src/CMakeLists.txt

```diff
@@ -1,8 +1,22 @@
-add_library(sqlpp-mysql
-		connection.cpp
-		prepared_statement.cpp
-		char_result.cpp
-		bind_result.cpp
-		detail/connection_handle.cpp)
+
+set(sqlpp_mysql_SRCs
+	connection.cpp
+	prepared_statement.cpp
+	char_result.cpp
+	bind_result.cpp
+	detail/connection_handle.cpp
+)
+
+add_library(sqlpp-mysql ${sqlpp_mysql_SRCs})
+add_library(sqlpp11::mysql ALIAS sqlpp-mysql)
+
+target_link_libraries(sqlpp-mysql
+	INTERFACE date::date mariadb
```

Comment:
I'm slightly confused. Why mariadb? Doesn't the project use pure mysql? 

And are date and mariadb (mysql) really just interface dependencies? Not private dependencies?

Reply:
Maybe I can use mysqld instead of mariadb, which would be libmysqld, but as suggested by the file cmake / FindMySQL.cmake on line 16 I decided to use the same name.

About dependencies, I believe they can be INTERFACE here!

Reply:
Oh I just see what happened here, I think. So these names like "mariadb" or "libmysqld" is how the libaries are actually called. Or at least a hint how they called ... depending on the platform cmake will then search for libmariadb for example. 

What you usually want to do is to link against cmake targets (like date::data ... or sqlpp11). For instance Sqlite3 defines a cmake target: 
https://cmake.org/cmake/help/latest/module/FindSQLite3.html

You cann see in the source code of the FindSqlite3.cmake Script (which is maintained by kitware) at the end that they declare such a cmake target: 
https://github.com/Kitware/CMake/blob/master/Modules/FindSQLite3.cmake#L63

The mysql.cmake script here doesn't do this. So "mariadb" is not any cmake target. If cmake doesn't find a cmake target named like this, it will search directly for a library named like this. Just to test this, comment out following lines: 

find_package(MySql REQUIRED)     (36 in CMakeLists.txt) 
include_directories("${MYSQL_INCLUDE_DIRS}") (46 in CMakeLists.txt) 

So basically don't call the FindMysql.cmake Script anymore. You will see that it will still find the library mariadb. That's because you're on Linux and Linux puts all libraries in a central place, so it finds the library without any extra help :-)

Take a guess what happens on Windows? Take a look in findMySql.cmake 

```
	find_library(MYSQL_LIBRARY
		NAMES libmariadb
		PATH_SUFFIXES lib
		PATHS "$ENV{SystemDrive}/Program Files/mariadb-connector-c/lib/mariadb"
					"$ENV{SystemDrive}/Program Files (x86)/mariadb-connector-c/lib/mariadb"
		)
```

As you see the lib has a different Name, but it also needs some extra hints where you find it. So that's the reason why the code currently fails on Windows (see pipeline). 

You basically have two options: 
1. Just use ${MYSQL_LIBRARY}  ... that's basically the library name put in a variable by the script, which will be different depending on your platform (On Linux it will just be mariadb) 
1. Actually add a mysql cmake target similar like here: https://github.com/Kitware/CMake/blob/master/Modules/FindSQLite3.cmake#L63   ... that also has the advantage that you can add the mysql include directory here directly and don't need line 46 in the main script (include_directories("${MYSQL_INCLUDE_DIRS}")). So that's definitely the better way, but it's slightly more complitcated

I hope you understand my explanations. Modern CMake is actually rather easy, but it's difficult if you have libaries like MySQL, which don't provide good Cmake Integration.

Reply:
Regarding the INTERFACE thing. It compiles fine because of different reasons. 

For the date lib:  The reason this doesn't fail is because sqlpp11 has date as a PUBLIC Dependency and so the connector will have it as well automatically. Otherwise the compiler would explain about char_result.cpp for instance, because you include "data/date.h" there. I would suggest just removing date here as it's already provided by the main lib, no need to duplicate it

For mysql: The reason has to do with what I wrote above. The lib here is no cmake target and as such bring include directories.  The CmakeLists.txt has the line  `include_directories("${MYSQL_INCLUDE_DIRS}")`, which will always include the header directories no matter if you put mysql here private or public or interface. And that's the reason it doesn't complain here. 
I would suggest to make mysql PUBLIC here as this is what makes logical sense. It would also be required if you follow my advice and make mysql a proper cmake target.